### PR TITLE
[Merged by Bors] - replace util weight with fixed float lib in tortoise

### PR DESCRIPTION
**Pull request by dshulyak**
related: https://github.com/spacemeshos/go-spacemesh/issues/3763

- ballot weight is computed with rational numbers (using bit.Rat), so that space unit can be encoded without loss of precision
- tortoise get space unit in big.Rat and uses spacemeshos/fixed library for math and comparison with threshold. i think it is safe to consider this as an implementation detail and can be safely swapped to anything more efficient in future
- util.Weight is removed

### CODE REVIEW

File: tortoise/verifying.go [hunk 1]
```diff
@@ -76,13 +75,13 @@ func (v *verifying) verify(logger log.Log, lid types.LayerID) bool {
...
 
-	margin := util.WeightFromUint64(0).
-		Add(v.totalGoodWeight).
+	margin := v.totalGoodWeight.
 		Sub(layer.verifying.goodUncounted)
 	uncounted := v.expectedWeight(v.Config, lid).
 		Sub(margin)
-	if uncounted.Sign() > 0 {
-		margin.Sub(uncounted)
+	// GreaterThan(zero) returns true even if value with negative sign
```
Comment (countvonzero) [inline tortoise/verifying.go:82]: curious. is this a bug in fixed64 library?

File: tortoise/tortoise.go [hunk 2]
```diff
@@ -694,12 +695,16 @@ func (t *turtle) onAtx(atx *types.ActivationTxHeader) {
 			log.Uint64("weight", atx.GetWeight()),
 		)
 		epoch.atxs[atx.ID] = atx.GetWeight()
-		epoch.weight += atx.GetWeight()
+		if atx.GetWeight() > math.MaxInt64 {
+			// atx weight is not expected to overflow int64
+			panic("fixme: atx size overflows int64 " + strconv.FormatUint(atx.GetWeight(), 10))
```
Comment (countvonzero) [inline tortoise/tortoise.go]: log.Fatal

File: tortoise/tortoise.go [hunk 3]
```diff
@@ -694,12 +695,16 @@ func (t *turtle) onAtx(atx *types.ActivationTxHeader) {
...
+		}
+		epoch.weight = epoch.weight.Add(fixed.New64(int64(atx.GetWeight())))
```
Comment (countvonzero) [inline tortoise/tortoise.go:701]: it is also expected that total epoch weight will not exceed uint64. does `fixed.New64` satisfy that?

**Review by countvonzero — APPROVED** [review]

File: tortoise/tortoise.go (hunk 3 above)
Comment (dshulyak) [reply tortoise/tortoise.go:701]: yes, it checks overflow on math operations

File: tortoise/verifying.go (hunk 1 above)
Comment (dshulyak) [reply tortoise/verifying.go:82]: i want to understand that as well)

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**